.github/workflows/cross-compile.yml

```diff
@@ -0,0 +1,177 @@
+name: Cross-compile native libraries
+on: workflow_dispatch
+
+env:
+  MAVEN_OPTS: -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+  GITHUB_BOT_NAME: github-actions
+  GITHUB_BOT_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
+
+jobs:
+  create-branch:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - run: echo "precompiled_branch=$(git branch --show-current)-precompiled-natives" >> $GITHUB_ENV
+      - run: echo "rev=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+      - name: Create branch for precompiled natives
+        run: git checkout -b ${{ env.precompiled_branch }} && git push --force origin HEAD
+    outputs:
+      precompiled_branch: ${{ env.precompiled_branch }}
+      base_rev: ${{ env.rev }}
+
+  linux-windows:
+    runs-on: ubuntu-20.04
+    needs: create-branch
+    strategy:
+      matrix:
+        include:
+          - target: linux_32
+            profile: x86
+            image: ubuntu:14.04
+            packages: cmake3 g++-multilib
+
+          - target: linux_64
+            profile: x86_64
+            image: ubuntu:14.04
+            packages: cmake3 g++
+
+          - target: linux_arm
+            profile: armhf
+            image: ubuntu:14.04
+            packages: cmake3 g++-arm-linux-gnueabihf
+
+          - target: linux_arm64
+            profile: aarch64
+            image: ubuntu:14.04
+            packages: cmake3 g++-aarch64-linux-gnu
+
+          - target: linux_ppc
+            profile: ppc64
+            image: ubuntu:14.04
+            packages: cmake3 g++-powerpc64le-linux-gnu
+
+          - target: windows_32
+            profile: mingw32
+            image: ubuntu:20.04
+            packages: file cmake mingw-w64 binutils-mingw-w64-i686
+
+          - target: windows_64
+            profile: mingw64
+            image: ubuntu:20.04
+            packages: file cmake mingw-w64
+
+          - target: windows_arm64
+            profile: mingwaarch64
+            image: ubuntu:20.04
+            packages: file cmake clang
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Pull docker images
+        run: docker pull ${{ matrix.image }}
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt
```

Comment:
Does this need to change with the rebranding of adopt->adoptium or is this ok?

Reply:
That's one of few options provided by plugin: https://github.com/actions/setup-java#supported-distributions , so it should be ok.

Reply:
It appears they just need to update their [JSON files](https://github.com/actions/setup-java/blob/b53500dabc37db6cc1dcd35059bdcd85df25d140/__tests__/data/adopt.json#L659)... 11.0.12 was released [this week](https://github.com/adoptium/temurin11-binaries/releases/jdk-11.0.12%2B7/) at the new project home.

Reply:
Switched to `temurin`. Does not work ATM, should be ready by [monday](https://github.com/actions/setup-java/pull/201)